docs/commands.md

```diff
@@ -0,0 +1,153 @@
+
+The value(s) provided for image and repository are persisted to the `.faas.yaml` file so that subsequent invocations do not require the user to specify these again.
+
+Similar `kn` command: none.
+
+```console
+faas build [-i <image> -r <repository> -p <path>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas build [-i <image> -r <repository> -p <path>]
+```
+
+## `run`
+
+Runs the Function project locally in the container. If a container has not yet been created, prompts the user to run `faas build`. Note: there is no option to specify a path to the project.
+
+Similar `kn` command: none.
+
+```console
+faas run
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas run
+```
+
+## `deploy`
+
+Deploys the Function project in the current directory. The user may specify a path to the project directory as a flag. Reads the `.faas.yaml` configuration file to determine the image name. Derives the service name from the project name. There is no command line option to specify the image name, although this can be changed in `.faas.yaml`. There is no mechanism by which the user can specify the service name. By default, the image is deployed to the namespace corresponding to the value specified in `.faas.yaml`, or if unset to the currently active Kubernetes namespace. This can be changed with a command line flag. If changed, this value will be persisted in `.faas.yaml`.
+
+Similar `kn` command: `kn service create NAME --image IMAGE [flags]`. This command allows a user to deploy a Knative Service by specifying an image, typically one hosted on a public container registry such as docker.io. The deployment options which the `kn` command affords the user are quite broad. The `kn` command in this case is quite effective for a power user. The `faas deploy` command has a similar end result, but is definitely easier for a user just getting started to be successful with.
+
+```console
+faas deploy [-n <namespace> -p <path>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas deploy [-n <namespace> -p <path>]
+```
+
+## `update`
+
+Updates the deployed Function project in the current directory. The user may specify the path on the command line with a flag. Reads the `.faas.yaml` configuration file to determine the image name. Derives the service name from the project name. The deployed Function is updated with a new container image that is pushed to a user repository, and the Knative `Service` is then updated.
+
+The namespace defaults to the value in `.faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `.faas.yaml`. The user may specify a repository on the command line.
```

Comment:
This is a really helpful paragraph.  Perhaps it should be repeated in the `deploy` section above as well.